electron_app/src/electron-main.js

```diff
@@ -214,6 +229,155 @@ ipcMain.on('ipcCall', async function(ev, payload) {
     });
 });
 
+ipcMain.on('seshat', async function(ev, payload) {
+    if (!mainWindow) return;
+
+    const sendError = (id, e) => {
+        const error = {
+            message: e.message
+        }
+
+        mainWindow.webContents.send('seshatReply', {
+            id:id,
+            error: error
+        });
+    }
+
+    const args = payload.args || [];
+    let ret;
+
+    switch (payload.name) {
+        case 'supportsEventIndexing':
+            if (Seshat === null) ret = false;
+            else ret = true;
+            break;
+
+        case 'initEventIndex':
+            if (eventIndex === null) {
+                let p = path.normalize(eventStorePath);
+                try {
+                    await afs.mkdir(p, {recursive: true});
+                    eventIndex = new Seshat(p, {passphrase: "DEFAULT_PASSPHRASE"});
+                } catch (e) {
+                    sendError(payload.id, e);
+                    return;
+                }
+            }
+            break;
+
+        case 'closeEventIndex':
+            eventIndex = null;
+            break;
+
+        case 'deleteEventIndex':
+            const deleteFolderRecursive = async(p) =>  {
+                for (let entry of await afs.readdir(p)) {
+                    const curPath = path.join(p, entry);
+                    await afs.unlink(curPath);
+                }
+            }
+
+            try {
+                await deleteFolderRecursive(path.normalize(eventStorePath));
+            } catch (e) {
```

Comment:
Maybe `sendError` to match the others...?

Reply:
Not sure the other side cares about this error. It will mostly fire if there isn't a folder to delete in the first place.